### PR TITLE
[Patterns]: Add new `Header Centered Search` pattern

### LINKED ISSUE #9035: Create `Header: Centered Search` pattern
Centered Search

Check Figma designs F5AAbRJOGbTeEweXPFtHgf-fi-1540_33086 for the latest updates and mobile views.

### PR DESCRIPTION
**Pull request by albarin**
This PR implements the `Header Centered Search` pattern.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9035

#### Design
Centered Search

#### Implementation
Screenshot 2023-04-18 at 10 15 38

#### User-Facing Testing

1. Create a new page or post
2. Make sure the `WooCommerce Header Centered Search` pattern appears under the WooCommerce category dropdown.
3. Insert in and make sure it shows as expected on the design.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: patterns/header-centered-search-pattern.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
+			<!-- /wp:spacer -->
+
+			<!-- wp:spacer -->
+			<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
+			<!-- /wp:spacer -->
+
+			<!-- wp:spacer -->
+			<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
+			<!-- /wp:spacer -->
```
Comment (kmanijak) [inline patterns/header-centered-search-pattern.php]: What's the purpose of multiple Spacer blocks between Search and Customer Account? Could they be removed? I did that in Editor and I think the layout stayed the same.

**Review by kmanijak — COMMENTED** [review]
I can find the pattern and apply it, however the search seems to by squeezed:
| Editor | image |
| --- | --- |
| Frontend | image |

As you can see the placeholder doesn't show up fully even though there's a space for it. I tried to use the width control, but that doesn't help and seems buggy (I don't consider it a part of this PR though, just wanted to show how it behaves).
https://user-images.githubusercontent.com/20098064/232742699-9334cee0-030c-4e2f-9c39-037e0f710744.mov

Do you think there are additional steps/conditions that I should follow/satisfy?

File: patterns/header-centered-search-pattern.php (hunk 1 above)
Comment (albarin) [reply patterns/header-centered-search-pattern.php]: Yeah, it's not a good solution, I wanted to add some margin between the search block and the customer account block. I've added support for `margin` on the `Customer Account` block, and added some left margin to it.

**Comment by albarin** [comment]
@kmanijak yep, that's weird, I've pushed some updates, I'm trying with a width in `px`, wdyt? 🤔

File: patterns/header-centered-search-pattern.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+		<!-- wp:group {"style":{"spacing":{"blockGap":"8px"},"typography":{"fontSize":"13px"}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"right"}} -->
+		<div class="wp-block-group" style="font-size:13px">
+			<!-- wp:group {"style":{"spacing":{"blockGap":"16px"}},"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+			<div class="wp-block-group">
+				<!-- wp:separator {"className":"is-style-wide"} -->
+				<hr class="wp-block-separator has-alpha-channel-opacity is-style-wide" />
+				<!-- /wp:separator -->
+			</div>
+			<!-- /wp:group -->
```
Comment (kmanijak) [inline patterns/header-centered-search-pattern.php]: The only thing I found, was that this Row with Separator inside doesn't influence the look of the template. Could they be removed or there's some other reason that I missed?

| With | Screen Shot 2023-04-19 at 08 58 16 AM |
| --- | --- |
| Without | image |

**Review by kmanijak — APPROVED** [review]
That tests well! I left one comment about a Row with Separator, but pre-approving! 👍

**Comment by kmanijak** [comment]
> @kmanijak yep, that's weird, I've pushed some updates, I'm trying with a width in px, wdyt? 🤔

I like the idea, because there's a handle on the right and merchant can easily adjust the width in case they need it, so for me, it's a win-win:
image

File: patterns/header-centered-search-pattern.php (hunk 2 above)
Comment (albarin) [reply patterns/header-centered-search-pattern.php]: Thanks, removed!

**Comment by albarin** [comment]
@shaunandrews this is ready for you to take a look, also any suggestions for the pattern title? Thanks! 😄

**Comment by shaunandrews** [comment]
CleanShot 2023-04-19 at 12 06 47@2x


I think the block spacing between the logo and title could be reduced down by half to 20px.

--

CleanShot 2023-04-19 at 12 05 53@2x

The search looks strange to me. Almost like it's not rendering the right CSS. This is with the Tsubaki theme.

--

CleanShot 2023-04-19 at 12 07 10@2x

On the Navigation block I changed the Font Appearance to "Normal" and found that things look much better that the bolded links I saw at first. Is this is theme thing?

--

For the Search placeholder lets reduce it down to simply "Search our store".

**Comment by shaunandrews** [comment]
Oh, and for a name, how about "Centered Menu with Search"?

**Comment by albarin** [comment]
@shaunandrews I addressed all your feedback.
I think the solution I did for the search input border is maybe too hacky, so let's discuss it with the team before merging.

**Comment by thealexandrelara** [comment]
WooCommerce Blocks 10.1.0 will be released soon. I'm bumping this PR to the next milestone

**Comment by shaunandrews** [comment]
CleanShot 2023-04-24 at 12 27 00@2x

While clever, I don't think using the HTML block is the right solution; It adds the content to the canvas in a confusing way.

The border comes from the block itself: https://github.com/WordPress/gutenberg/blob/fd98542335ce1a2eb0517b9e4f285d750b4c4645/packages/block-library/src/search/style.scss#L59

Its unclear to me why, though.

I think a potential solution here is to update the Tsubaki theme or the Search block.

**Comment by albarin** [comment]
> While clever, I don't think using the HTML block is the right solution; It adds the content to the canvas in a confusing way.

@shaunandrews Agree, I pushed another solution I thought today: instead of using `Button inside` use `Button outside` and adding the border outside with a stack element, I think it works. Let me know what you think 😄

**Pull request merged**